File: Examples/VisualBasic/3DModeling/Primitive3DModels.vb
Imports System.Collections
Imports Aspose.ThreeD
Imports Aspose.ThreeD.Animation
Imports Aspose.ThreeD.Entities
Imports Aspose.ThreeD.Formats

Namespace _3DModeling
    Class Primitive3DModels
        Public Shared Sub Run()
            ' ExStart:Primitive3DModels
            ' The path to the documents directory.
            Dim MyDir As String = RunExamples.GetDataDir()

            ' Initialize a Scene object
            Dim scene As New Scene()
            ' Create a Box model
            scene.RootNode.CreateChildNode("box", New Box())
            ' Create a Cylinder model
            scene.RootNode.CreateChildNode("cylinder", New Cylinder())
            ' Save drawing in the FBX format
            MyDir = MyDir & RunExamples.GetOutputFilePath("test.fbx")
            scene.Save(MyDir, FileFormat.FBX7500ASCII)

            ' ExEnd:Primitive3DModels
            Console.WriteLine(Convert.ToString(vbLf & "Building a scene from primitive 3D models successfully." & vbLf & "File saved at ") & MyDir)
        End Sub
    End Class
End Namespace

File: Examples/CSharp/3DModeling/Primitive3DModels.cs
using System;
using System.IO;
using System.Collections;
using Aspose.ThreeD;
using Aspose.ThreeD.Animation;
using Aspose.ThreeD.Entities;
using Aspose.ThreeD.Formats;

namespace Aspose._3D.Examples.CSharp._3DModeling
{
    class Primitive3DModels
    {
        public static void Run()
        {
            // ExStart:Primitive3DModels
            // The path to the documents directory.
            string MyDir = RunExamples.GetDataDir();

            // Initialize a Scene object
            Scene scene = new Scene();
            // Create a Box model
            scene.RootNode.CreateChildNode("box", new Box());
            // Create a Cylinder model
            scene.RootNode.CreateChildNode("cylinder", new Cylinder());
            // Save drawing in the FBX format
            MyDir = MyDir + RunExamples.GetOutputFilePath("test.fbx");
            scene.Save(MyDir, FileFormat.FBX7500ASCII);

            // ExEnd:Primitive3DModels
            Console.WriteLine("\nBuilding a scene from primitive 3D models successfully.\nFile saved at " + MyDir);
        }
    }
}
